private/bufpkg/bufimage/bufimage.go

```diff
@@ -480,7 +481,7 @@ func ImageWithoutImports(image Image) Image {
 			newImageFiles = append(newImageFiles, imageFile)
 		}
 	}
-	return newImageNoValidate(newImageFiles)
+	return newImageNoValidate(newImageFiles, image.Resolver())
```

Comment:
This was a separate bug: this function (`ImageWithoutImports`) was creating an image that had no resolver.